cloud/scope/lbs_load_balancer_reconciler.go

```diff
@@ -0,0 +1,326 @@
+	// 	CompartmentId: common.String(s.GetCompartmentId()),
+	// 	DisplayName:   common.String(lb.Name),
+	// 	SortBy:        loadbalancer.ListLoadBalancersSortByTimecreated,
+	// 	SortOrder:     loadbalancer.ListLoadBalancersSortOrderDesc,
+	// })
+	// if err != nil {
+	// 	s.Logger.Error(err, "failed to list lbs after creation")
+	// 	return nil, nil, errors.Wrap(err, "failed to list lbs after creation")
+	// }
+
+	// if len(listResp.Items) < 1 {
+	// 	s.Logger.Error(err, "failed to find lbs apiserver lb after creation")
+	// 	return nil, nil, errors.Wrap(err, "failed to find lbs apiserver lb after creation")
+	// }
+
+	lbs, err := s.LoadBalancerServiceClient.GetLoadBalancer(ctx, loadbalancer.GetLoadBalancerRequest{
+		LoadBalancerId: wr.LoadBalancerId,
+	})
+	if err != nil {
+		s.Logger.Error(err, "failed to get apiserver lb after creation")
+		return nil, nil, errors.Wrap(err, "failed to get apiserver lb after creation")
+	}
+
+	lbIp, err := s.getLbsLoadbalancerIp(lbs.LoadBalancer)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	s.Logger.Info("Successfully created apiserver lb", "lb", lbs.Id, "ip", lbIp)
+	return lbs.Id, lbIp, nil
+}
+
+func (s *ClusterScope) getLbsLoadbalancerIp(lb loadbalancer.LoadBalancer) (*string, error) {
+	var lbIp *string
+	if len(lb.IpAddresses) < 1 {
+		return nil, errors.New("lb does not have valid ip addresses")
+	}
+	if *lb.IsPrivate {
+		lbIp = lb.IpAddresses[0].IpAddress
+	} else {
+		for _, ip := range lb.IpAddresses {
+			if *ip.IsPublic {
+				lbIp = ip.IpAddress
+			}
+		}
+	}
+	if lbIp == nil {
+		return nil, errors.New("lb does not have valid public ip address")
+	}
+	return lbIp, nil
+}
+
+// IsLBEqual determines if the actual loadbalancer.LoadBalancer is equal to the desired.
+// Equality is determined by DisplayName, FreeformTags and DefinedTags matching.
+func (s *ClusterScope) IsLbsLBEqual(actual *loadbalancer.LoadBalancer, desired infrastructurev1beta2.LoadBalancer) bool {
+	if desired.Name != *actual.DisplayName {
+		return false
+	}
+	return true
+}
+
+// GetLbsLoadBalancer retrieves the Cluster's loadbalancer.LoadBalancer using the one of the following methods
+//
+// 1. the OCICluster's spec LoadBalancerId
+//
+// 2. Listing the LoadBalancers for the Compartment (by ID) and DisplayName then filtering by tag
+func (s *ClusterScope) GetLbsLoadBalancer(ctx context.Context) (*loadbalancer.LoadBalancer, error) {
+	lbOcid := s.OCIClusterAccessor.GetNetworkSpec().APIServerLB.LoadBalancerId
+	if lbOcid != nil {
+		resp, err := s.LoadBalancerServiceClient.GetLoadBalancer(ctx, loadbalancer.GetLoadBalancerRequest{
+			LoadBalancerId: lbOcid,
+		})
+		if err != nil {
+			return nil, err
+		}
+		nlb := resp.LoadBalancer
+		if s.IsResourceCreatedByClusterAPI(nlb.FreeformTags) {
+			return &nlb, nil
+		} else {
+			return nil, errors.New("cluster api tags have been modified out of context")
+		}
+	}
+	lbs, err := s.LoadBalancerServiceClient.ListLoadBalancers(ctx, loadbalancer.ListLoadBalancersRequest{
```

Comment:
since awe are doing this new, we should add pagination here, you can check for some places where we added pagination

Reply:
@shyamradhakrishnan   I'm not quite sure about this part. I know there is a `page` variable in the ListLoadBalancersRequest, but I don't see how it's used in the current NLB functions.